src/compiler/program.ts

```diff
@@ -991,7 +991,13 @@ namespace ts {
         return sortAndDeduplicateDiagnostics(diagnostics);
     }
 
-    export function formatDiagnostics(diagnostics: Diagnostic[], host: CompilerHost): string {
+    export interface FormatDiagnosticsHost {

```

Comment:
It's a subset of CompilerHost/sys functionality - specifically the minimal subset needed to properly format diagnostics. Like how use use ModuleResolutionHost to indicate where we need the minimal functionality for resolving modules. Grouping the methods needed into a "host" like this is simply cleaner than passing in a collection of arguments or closures directly.